### PR TITLE
refactor: fetch exchange rates only for live coins

### PR DESCRIPTION
**Pull request by dated**
## Summary

Right now the wallet fetches the exchange rate for all coins, live or not. With these change the respective methods return early instead of fetching the rate for non-live coins.

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: packages/platform-sdk-profiles/src/wallets/wallet.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ export class Wallet implements ReadWriteWallet {
 	}
 
 	public convertedBalance(): BigNumber {
-		if (this.network().isTest()) {
+		if (!this.network().isLive()) {
```
Comment (faustbrian) [inline packages/platform-sdk-profiles/src/wallets/wallet.ts:172]: Why is this necessary? If the `isTest` check doesn't work then `isTest` is bugged.

File: packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts [hunk 2]
```diff
@@ -29,6 +29,10 @@ export class ExchangeRateService {
 				.filter((wallet: ReadWriteWallet) => wallet.currency() === currency);
 		}
 
+		if (!wallets.length || !wallets[0].network().isLive()) {
```
Comment (faustbrian) [inline packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts:32]: Can't we already filter out `isTest` wallets before we even filter the currency? Then the second filter would be a noop if there are no live wallets for that currency.

File: packages/platform-sdk-profiles/src/wallets/wallet.ts (hunk 1 above)
Comment (dated) [reply packages/platform-sdk-profiles/src/wallets/wallet.ts:172]: not necessary at all, changing back.

File: packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts [hunk 3]
```diff
@@ -26,7 +26,13 @@ export class ExchangeRateService {
 			wallets = profile
 				.wallets()
 				.values()
-				.filter((wallet: ReadWriteWallet) => wallet.currency() === currency);
+				.filter((wallet: ReadWriteWallet) => wallet.currency() === currency && !wallet.network().isTest());
```
Comment (faustbrian) [inline packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts:29]:
```suggestion
				.filter((wallet: ReadWriteWallet) => wallet.currency() === currency && wallet.network().isLive());
```

File: packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts [hunk 4]
```diff
@@ -26,7 +26,13 @@ export class ExchangeRateService {
...
+		} else {
+			wallets = wallets.filter((wallet: ReadWriteWallet) => !wallet.network().isTest());
```
Comment (faustbrian) [inline packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts:31]:
```suggestion
			wallets = wallets.filter((wallet: ReadWriteWallet) => wallet.network().isLive());
```

File: packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts (hunk 2 above)
Comment (dated) [reply packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts]: Right now the methods for the exchange rate are based on `allByCoin`, whose purpose is questionable at this point, as it groups live & test wallets together.
